setup.py

```diff
@@ -364,19 +421,29 @@ def build_extension(self, ext: _BaseExtension) -> None:
         src_file: Path = ext.src_path(self)
```

Comment:
The whole purpose of InstallerBuildExt and BuiltExtension is that it installs files that are built by CMake. Since you're installing files that are not built that way, you should use a different installation type instead of saying "this built file is not actually built".

Extending CustomBuildPy would be a better fit: its purpose is to copy non-built, platform-agnostic files. You won't be able to use ext_modules entrie; instead you embed the file list in CustomBuildPy like we do with the .fbs files. Doing this should dramatically simplify this PR: all of the changes would be isolated to CustomBuildPy, and it can do the globbing in a header-specific loop. It wouldn't need to change any of the "built file" logic.

In the future we could define a custom pyproject.toml section that contains the .fbs and header mappings, to avoid embedding them in the builder code. But it should be fine for now.